File: 1482-how-many-numbers-are-smaller-than-the-current-number/how-many-numbers-are-smaller-than-the-current-number.py
class Solution:
    def smallerNumbersThanCurrent(self, nums: List[int]) -> List[int]:
        sortedNums = sorted(nums)
        hash = {}
        output = []
        for i in range(len(sortedNums)):
            if sortedNums[i] not in hash:
                hash[sortedNums[i]] = i
        
        for each in nums:
            output.append(hash[each])
        
        return output

            
        

File: 1482-how-many-numbers-are-smaller-than-the-current-number/how-many-numbers-are-smaller-than-the-current-number.js
var smallerNumbersThanCurrent = function(nums) {
    hash = {};
    output = [];
    sortedNums = [...nums];
    sortedNums.sort((a,b)=>a-b);
    
    for (let i = 0;i < nums.length;i++){
        if (!(sortedNums[i] in hash)){
            hash[sortedNums[i]] = i;
        }
    }
    for (each of nums){
        output.push(hash[each]);
    }
    return output;
    
};